frontend/lib/resolution/Resolver.cpp

```diff
@@ -2147,6 +2147,16 @@ bool Resolver::resolveSpecialNewCall(const Call* call) {
   auto newExpr = call->calledExpression()->toNew();
   auto& re = byPostorder.byAst(call);
   auto& reNewExpr = byPostorder.byAst(newExpr);
+
+  // Rewrite 'new dmap' to 'new _distribution'
+  if (auto rt = reNewExpr.type().type()->toRecordType()) {
+    if (rt->id().symbolPath() == "ChapelArray.dmap") {
+      reNewExpr.setType(QualifiedType(
+          reNewExpr.type().kind(), CompositeType::getDistributionType(context),
+          reNewExpr.type().param()));
+    }
+  }
+
```

Comment:
I think it would work, but why is it be better one way or the other? I think my leaning towards this spot was because it's before the guards for unknown or erroneous type, and because I felt like rewriting the type should happen as early as possible before further processing.

Reply:
I don't feel strongly enough to argue -- if you've considered my suggestion and think this way is better, I trust your judgement 👍 .

Reply:
Ok, that works for me. FWIW I was genuinely asking for reasoning why one way or the other is better, since I also don't feel strongly about it and felt like it was a toss up while writing it

Reply:
Oh gotcha, I buy that reasoning. I'll change it. It has the additional upside of being simpler since we just adjust a `Type*` rather than a whole `QualifiedType`.